packages/core/src/connectors/injected.ts

```diff
@@ -150,7 +150,7 @@ export function injected(parameters: InjectedParameters = {}) {
       if (!isReconnecting) {
         accounts = await this.getAccounts().catch(() => null)
         const isAuthorized = !!accounts?.length
-        if (isAuthorized)
+        if (isAuthorized && !shimDisconnect)
```

Comment:
```suggestion
        if (isAuthorized && shimDisconnect)
```

shouldn’t this be inversed as per https://github.com/wevm/wagmi/issues/3603#issuecomment-1948946993?

Reply:
You are right, I was confused by bug I've just found: 
`injected` is called twice: first by user code when creating `wagmiConfig`: `...injected({shimDisconnect: false})...`
Second here: https://github.com/wevm/wagmi/blob/a91c0b64ba8b3e6537a560e69724eb601f26af27/packages/core/src/createConfig.ts#L115
The bug is that when it's called for the second time, `shimDisconnect` is not passed, and it's set as `true` here: https://github.com/wevm/wagmi/blob/a91c0b64ba8b3e6537a560e69724eb601f26af27/packages/core/src/connectors/injected.ts#L91
So your code is right, but it won't help because of the bug I've just described.